addons/mail/models/ir_attachment.py

```diff
@@ -34,3 +34,15 @@ def register_as_main_attachment(self, force=True):
                     related_record.message_main_attachment_id = self
                 except AccessError:
                     pass
+
+    def _attachment_format(self, main_attachment = None, safari = None):
+        return [{
+            'checksum': attachment.checksum,
+            'id': attachment.id,
+            'filename': attachment.name,
+            'name': attachment.name,
+            'mimetype': 'application/octet-stream' if safari and attachment.mimetype and 'video' in attachment.mimetype else attachment.mimetype,
+            'is_main': main_attachment == attachment,
```

Comment:
I wouldn't mind if `main_attachment` became a result of message format itself, and if this key didn't exist on each individual attachment result.